scenes/menus/intro/components/intro.gd

```diff
@@ -12,6 +12,10 @@ var introduction: DialogueResource = preload("res://scenes/menus/intro/component
 
 
 func _ready() -> void:
+	# Wait for the scene transition from the splash screen to be complete.
+	while Pause.is_paused(Pause.System.PLAYER_INPUT):
```

Comment:
This to me seems a little hacky.

Maybe we can add a signal in the `Transitions` node (`Transitions.gd`) that triggers once the transition finishes?I'm a signal called: `transition_finished(is_entering: bool)` where is_entering is true when the transition is entering the scene, and false when leaving.

Reply:
The Pause autoload already has a signal (I remember because I requested it in the review).

```
signal pause_changed(system: System, paused: bool)
```

Reply:
That signal is awaited in the following line. The problem is that it can't be unconditional because, if you run the scene directly, no transition is running. So unless I'm missing something it has to be conditional even if we add the suggested new signal to `Transitions`.

The other option I considered is to have a method that would be called on the scene by `Transitions`, but again the scene would have to know whether to expect that method call.

If we were using the built-in pause functionality, we could have the scene-switching code pause the new scene until the end of the transition...

Reply:
Actually I think it can be made to work if we have the Transitions singleton unconditionally emit that signal with paused=false for each system. I'll experiment tomorrow. And I think pausing the scene after it's constructed may be too late - _ready would have been called.

Reply:
> That signal is awaited in the following line. The problem is that it can't be unconditional because, if you run the scene directly, no transition is running. So unless I'm missing something it has to be conditional even if we add the suggested new signal to Transitions.

You are right! It's a tricky one.

Another option is to delay the dialog just a second after the scene starts. My thinking being that nothing important should happen in the first second you enter a scene. The transitions and everything that happen behind should just be decoration.

Of course, pausing could also be an option. We could add the pause system as a parameter to the methods `change_to_xxx_with_transition` so the caller can choose wich system should pause during transition.

Reply:
OK. In the hope of unblocking this I've replaced the "wait until unpaused" loop with an unconditional 1-second pause before triggering the dialogue. Actually I'm not convinced this is better because we now have the 1-second transition duration hardcoded in a second place; but equally I don't think it's worth exposing the default transition duration in `Transitions`; or doing something more fancy with signals. This works for now. I plan to refactor this script to also play the introduction to the Stella and the Missing Star StoryQuest, so I'm keen to move on :)